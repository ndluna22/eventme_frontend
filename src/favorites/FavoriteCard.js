import React from "react";
import { Link } from "react-router-dom";

import "./FavoriteCard.css";

/** Show limited information about a company
 *
 * Is rendered by FavoriteList to show a "card" for each favorite artist.
 *
 * Favoritelist -> FavoriteCard
 */

function FavoriteCard({ id, name, images }) {
  console.debug("FavoriteCard", name);

  return (
    <Link className="FavoriteCard card" to={`/artists/${id}`}>
      <div className="card-body">
        <h6 className="card-title">
          {name}
          {<img src={images} alt={name} className="float-right ml-5" />}
        </h6>
        <p>{/* <small>{}</small> */}</p>
      </div>
    </Link>
  );
}

export default FavoriteCard;
