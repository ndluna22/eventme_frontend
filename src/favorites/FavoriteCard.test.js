import React from "react";
import { render } from "@testing-library/react";
import FavoriteCard from "./FavoriteCard";
import { MemoryRouter } from "react-router";

it("matches snapshot with logo", function () {
  const { asFragment } = render(
    <MemoryRouter>
      <FavoriteCard id="K8vZ9171o-f" name="Rithm School" />
    </MemoryRouter>
  );
  expect(asFragment()).toMatchSnapshot();
});

it("matches snapshot without logo", function () {
  const { asFragment } = render(
    <MemoryRouter>
      <FavoriteCard id="K8vZ9171o-f" name="Algo School" />
    </MemoryRouter>
  );
  expect(asFragment()).toMatchSnapshot();
});
