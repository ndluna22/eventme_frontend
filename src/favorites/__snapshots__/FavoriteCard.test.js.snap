// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches snapshot with logo 1`] = `
<DocumentFragment>
  <a
    class="FavoriteCard card"
    href="/artists/K8vZ9171o-f"
  >
    <div
      class="card-body"
    >
      <h6
        class="card-title"
      >
        Rithm School
        <img
          alt="Rithm School"
          class="float-right ml-5"
        />
      </h6>
      <p />
    </div>
  </a>
</DocumentFragment>
`;

exports[`matches snapshot without logo 1`] = `
<DocumentFragment>
  <a
    class="FavoriteCard card"
    href="/artists/K8vZ9171o-f"
  >
    <div
      class="card-body"
    >
      <h6
        class="card-title"
      >
        Algo School
        <img
          alt="Algo School"
          class="float-right ml-5"
        />
      </h6>
      <p />
    </div>
  </a>
</DocumentFragment>
`;
